### PR TITLE
Async database

### PR DESCRIPTION
**Pull request by hedgehog098**
async and depend   XD

### CODE REVIEW

**Comment by wxiwnd** [comment]
Please solve the conflict first. That means you should rebase your branch.
And try to push one commit just for one feature or change.

File: kagami/routes/deps.py [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
 from fastapi import Depends
+from sqlalchemy.ext.asyncio import AsyncSession
 
 from ..core import Supervisor
+from ..database.database_service.engine import DatabaseEngine
 from ..server import supervisor
 
 SupervisorDeps = Annotated[Supervisor, Depends(lambda: supervisor)]
+
+SessionDeps = Annotated[AsyncSession, Depends(lambda: DatabaseEngine.session_factory)]
```
Comment (wxiwnd) [inline kagami/routes/deps.py]: DatabaseEngine.session_factory is a function, don't use lambda here.

Comment (wxiwnd) [reply kagami/routes/deps.py]: Better to use asynccontextmanager to manage the life cycle of session. If directly use session_factory here, every new request will create a new session and maybe this session will not quit correctly.

**Comment by wxiwnd** [comment]
https://github.com/ProjectSoftoid/Kagami/pull/18

File: kagami/routes/deps.py (hunk 1 above)
Comment (hedgehog098) [reply kagami/routes/deps.py]: This was solved by [0eb5c97](https://github.com/ProjectSoftoid/Kagami/pull/12/commits/0eb5c971118ab57e07b650e6f201996e1d786e69)

File: kagami/routes/deps.py [hunk 2]
```diff
@@ -1,8 +1,24 @@
...
 
 from ..core import Supervisor
+from ..database.engine import DatabaseEngine
 from ..server import supervisor
 
 SupervisorDeps = Annotated[Supervisor, Depends(lambda: supervisor)]
+
+
+@asynccontextmanager
+async def some_async_generator():
```
Comment (wxiwnd) [inline kagami/routes/deps.py]: The name can be changed into session_generator() or other words.

File: kagami/routes/deps.py [hunk 3]
```diff
@@ -1,8 +1,24 @@
...
+    session: AsyncSession
+    session = DatabaseEngine.session_factory
```
Comment (wxiwnd) [inline kagami/routes/deps.py]: This type hint can be wrote like:
session: AsyncSession = DatabaseEngine.session_factory()
Note: session_factory() is a function, so you have to call it.

Comment (wxiwnd) [reply kagami/routes/deps.py]: Another thing is you can add the return type hint for a function, like:
async def session_generator() -> AsyncSession | None, means this function return AsyncSession or None if Error occurs.

**Review by wxiwnd — COMMENTED** [review]
Control the lifespan is important, I think this is a good feature.

**Review by wxiwnd — APPROVED** [review]
LGTM

**Pull request merged**